OrbitCore/LinuxEventTracer.cpp

```diff
@@ -21,69 +22,78 @@
 #include "OrbitModule.h"
 #include "OrbitProcess.h"
 #include "Params.h"
-#include "PrintVar.h"
 #include "SamplingProfiler.h"
 #include "TimerManager.h"
 
-//-----------------------------------------------------------------------------
-void LinuxEventTracer::Start() {
-  PRINT_FUNC;
-#if __linux__
-  *m_ExitRequested = false;
-
-  m_Thread = std::make_shared<std::thread>(&LinuxEventTracer::Run, m_Pid,
-                                           m_Frequency, m_ExitRequested);
-  m_Thread->detach();
-#endif
-}
-
-//-----------------------------------------------------------------------------
-void LinuxEventTracer::Stop() { *m_ExitRequested = true; }
-
-//-----------------------------------------------------------------------------
-void LinuxEventTracer::Run(
-    pid_t a_Pid, int32_t a_Frequency,
+void LinuxEventTracerThread::Run(
     const std::shared_ptr<std::atomic<bool>>& a_ExitRequested) {
-  int cpus = static_cast<int>(std::thread::hardware_concurrency());
-  // Some compilers does not support hardware_concurrency.
-  if (cpus == 0) {
-    cpus = std::stoi(LinuxUtils::ExecuteCommand("nproc"));
+  if (!ComputeFrequency()) {
+    return;
   }
 
+  LoadNumCpus();
+
   std::vector<int32_t> fds;
+  std::unordered_map<int32_t, Function*> uprobe_fds_to_function;
+  std::unordered_map<int32_t, Function*> uretprobe_fds_to_function;
 
   if (GParams.m_TrackContextSwitches) {
     if (GParams.m_SystemWideScheduling) {
       // perf_event_open for all cpus to keep track of process spawning.
-      for (int cpu = 0; cpu < cpus; cpu++) {
+      for (int32_t cpu = 0; cpu < cpus_; cpu++) {
         fds.push_back(LinuxPerfUtils::context_switch_open(-1, cpu));
       }
     } else {
       // perf_event_open for all cpus and the PID to keep track of process
       // spawning.
-      fds.push_back(LinuxPerfUtils::context_switch_open(
-          Capture::GTargetProcess->GetID(), -1));
+      fds.push_back(LinuxPerfUtils::context_switch_open(pid_, -1));
+    }
+  }
+
+  LinuxPerfEventProcessor uprobe_event_processor(
+      std::make_unique<BpfTraceVisitor>());
```

Comment:
It confuses me that this is created with the BpfTraceVisitor, even when !m_UseBpftrace. Is this just because that's more convenient or is this also used when not using bpftrace?

Reply:
This is for convenience, I'm removing all references to BpfTrace in the next PR.